test/e2e/main_test.go

```diff
@@ -3,9 +3,77 @@ package e2e
 import (
 	"testing"
 
-	f "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/keycloak/keycloak-operator/pkg/apis"
+	keycloakv1alpha1 "github.com/keycloak/keycloak-operator/pkg/apis/keycloak/v1alpha1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
 )
 
+type deployedOperatorTestStep func(*testing.T, *framework.Framework, *framework.TestCtx, string) error
+type environmentInitializationStep func(*testing.T, *framework.Framework, *framework.TestCtx, string) error
+
+type CRDTest interface {
+	PrepareEnvironment() []environmentInitializationStep
+
+	GetSubsetOfTests() map[string]deployedOperatorTestStep
+}
+
 func TestMain(m *testing.M) {
-	f.MainEntry(m)
+	framework.MainEntry(m)
+
+}
+
+func TestKeycloakCRDS(t *testing.T) {
+	keycloakType := &keycloakv1alpha1.Keycloak{}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, keycloakType)
```

Comment:
This is tricky - I think adding types to scheme (which basically means deploying CRDs) should be part of environment setup.

Reply:
This seems to be necessary for running tests. Without it, the framework is not able to create Keycloak CR. This is running only once before all tests, not sure it is possible to add it somewhere else.

Error message:
```
no kind is registered for the type v1alpha1.Keycloak in scheme "github.com/operator-framework/operator-sdk/pkg/test/framework.go:127"
```

Reply:
I was thinking about putting it into `CRDTestStruct.prepareEnvironmentSteps`. But after thinking about this for a while, I guess you're right - it needs to be part of a "global" initialization.